es55x/src/main/scala/tech/beshu/ror/es/rrconfig/TransportRRConfigAction.scala

```diff
@@ -0,0 +1,121 @@
+import org.elasticsearch.transport.TransportService
+import tech.beshu.ror.configuration.loader.ComposedConfigLoader
+import tech.beshu.ror.configuration.loader.distributed.{NodeConfig, Timeout}
+import tech.beshu.ror.es.IndexJsonContentService
+import tech.beshu.ror.es.services.EsIndexJsonContentService
+import tech.beshu.ror.providers.{EnvVarsProvider, OsEnvVarsProvider}
+
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+class TransportRRConfigAction(setting: Settings,
+                              actionName: String,
+                              threadPool: ThreadPool,
+                              clusterService: ClusterService,
+                              transportService: TransportService,
+                              actionFilters: ActionFilters,
+                              env: Environment,
+                              indexContentProvider: IndexJsonContentService,
+                              nodeExecutor: String,
+                              indexNameExpressionResolver: IndexNameExpressionResolver,
+                              nodeResponseClass: Class[RRConfig],
+                              constructorDiscriminator: Unit)
+  extends TransportNodesAction[RRConfigsRequest, RRConfigsResponse, RRConfigRequest, RRConfig](
+    setting,
+    actionName,
+    threadPool,
+    clusterService,
+    transportService,
+    actionFilters,
+    indexNameExpressionResolver,
+    () => new RRConfigsRequest(),
+    () => new RRConfigRequest(),
+    nodeExecutor,
+    nodeResponseClass
+  ) {
+
+  import monix.execution.Scheduler.Implicits.global
+
+  implicit val envVarsProvider: EnvVarsProvider = OsEnvVarsProvider
+
+  @Inject
+  def this(setting: Settings,
+           actionName: String,
+           threadPool: ThreadPool,
+           clusterService: ClusterService,
+           transportService: TransportService,
+           actionFilters: ActionFilters,
+           env: Environment,
+           indexContentProvider: EsIndexJsonContentService,
+           indexNameExpressionResolver: IndexNameExpressionResolver,
+          ) =
+    this(
+      setting,
+      RRConfigAction.name,
+      threadPool,
+      clusterService,
+      transportService,
+      actionFilters,
+      env,
+      indexContentProvider,
+      ThreadPool.Names.GENERIC,
+      indexNameExpressionResolver,
+      classOf[RRConfig],
+      ()
+    )
+
+  override def newResponse(request: RRConfigsRequest, responses: util.List[RRConfig], failures: util.List[FailedNodeException]): RRConfigsResponse = {
+    new RRConfigsResponse(clusterService.getClusterName, responses, failures)
+  }
+
+  private def loadConfig() =
+    new ComposedConfigLoader(env.configFile(), indexContentProvider)
+      .load()
+      .map(_.map(_.map(_.raw)))
+
+  override def nodeOperation(request: RRConfigRequest): RRConfig = {
+    val nodeRequest = request.getNodeConfigRequest
+    val nodeResponse =
+      loadConfig()
```

Comment:
Even if it's not same, it will be in 5 sec. Were managing here configuration files, and index values. In other case we would present not fresh index value.

Reply:
what if user sets the refresh interval to 1h?

Reply:
ok, so let's recap:
* we have two types of config: from file loaded and from index loaded
* in Kibana plugin, we want to say: "hey, one/or more of you node/s has/have different configuration"
* when all nodes have an enabled in-index config, and if .readonlyrest index is a source of truth, there is not need to load a config by each node

I think we should think again:
* what we will gain with new solution (compared to the previous solution) when each node will have in-index config set?